src/trajectory.jl

```diff
@@ -231,13 +245,13 @@ isdivergent(
     nt::NUTS,
     H0::F,
     H′::F
-) where {F<:AbstractFloat} = (s.logu < nt.Δ_max + -H′) ? 0 : 1
+) where {F<:AbstractFloat} = Divergence(false, !(s.logu < nt.Δ_max + -H′))
```

Comment:
Perhaps we should use `!((-H0 < nt.Δ_max + -H′) && (s.logu < nt.Δ_max + -H′))` since `s.logu` might be bigger than `-H0`?

Reply:
This is what's in the original NUTS paper (Algorithm 6; https://arxiv.org/pdf/1111.4246.pdf). Seems that the divergence is check against the slice variable instead of the original point. Is this because we won't sample from the range between the original point and the slice variable any way?

Reply:
Need some attention from @yebai

Reply:
Both of the following 

- `s.logu < nt.Δ_max + -H′`
- `!((-H0 < nt.Δ_max + -H′) && (s.logu < nt.Δ_max + -H′))`

are correct in the context of Hamiltonian simulation, since they both preserve reversibility. The latter might be numerically more stable. I think we can keep using ``s.logu < nt.Δ_max + -H′` for now and revisit if it causes numerical stability.

Reply:
Wait a second, `logu` is equal to `log(rand(rng)) - H0` where `rand(rng)` is between `0` and `1` thus `log(rand(rng))` is less than `0`. So `logu` will not exceed `-H0` right?

Reply:
That's correct - I thought `logu=log(rand(rng))`. Sorry for the confusion.

Reply:
Hmm, not quite - `-H0` is also negative (since `H` is positive). So `logu` could exceed (i.e. become smaller than) `-H0`.

Reply:
Yes I checked it again and I believe whenever `-H0 < nt.Δ_max + -H′` is satisified, `s.logu < nt.Δ_max + -H′` would be satisfied as well.

`-H0 < nt.Δ_max + -H′` -> `-H0 - (-H′) < nt.Δ_max` -> `-H0 - (-H′) < nt.Δ_max - log(v)` (v ~ U(0, 1)) -> `log(v) -H0 - (-H′) < nt.Δ_max` ->  `logu - (-H′) < nt.Δ_max` -> `logu < nt.Δ_max -H′`.

So we either do `s.logu < nt.Δ_max + -H′` or `-H0 < nt.Δ_max + -H′` but we dont need the `&&` version. Is my understanding correct?

Reply:
> Yes I checked it again and I believe whenever -H0 < nt.Δ_max + -H′ is satisified, s.logu < nt.Δ_max + -H′ would be satisfied as well.

Indeed: `s.logu - nt.Δ_max = log(rand(rng)) - H0 - nt.Δ_max <= - H0 - nt.Δ_max < -H′`, so `s.logu <  nt.Δ_max + -H′`